### PR TITLE
CAMEL-13830: Remove examples folder

### PR DESCRIPTION
**Pull request by omarsmak**
This will remove the examples folder. However I had to remove the generation of the examples docs from the gulp file, perhaps @zregvart if you if have any hints on how I can manage the docs for the examples in the new [repo](https://github.com/apache/camel-examples/tree/master/examples#examples), would be highly appreciated :)

### CODE REVIEW

**Comment by zregvart** [comment]
Without the examples bit in the docs/gulpfile.js we'll loose the examples from the documentation. For example [this](https://camel.apache.org/manual/latest/testing.html#Testing-CamelTestExample) chapter would not have the example in it.

Antora requires that the examples are within the module's [examples](https://docs.antora.org/antora/2.2/modules/#examples-dir) directory.

We need to either commit the examples we currently have or find a way to pull them in the build from the `camel-examples` repository.

**Comment by omarsmak** [comment]
@zregvart indeed is going to be pretty tricky, perhaps we can do that in jenkins? Or use a git submodule? 
Anyway, I have committed back the deleted examples for the docs, at least in the meantime we have them. Can you please take a look if didn't mess anything with the docs? Thanks

File: docs/gulpfile.js [hunk 1]
```diff
@@ -195,15 +195,15 @@ function createComponentExamples() {
...
 );
 const nav = parallel(createComponentNav, createUserManualNav);
 const examples = series(deleteExamples, createUserManualExamples, createComponentExamples);
 
 exports.symlinks = symlinks;
 exports.nav = nav;
-exports.examples = examples;
-exports.default = series(symlinks, nav, examples);
-
+//exports.examples = examples;
```
Comment (zregvart) [inline docs/gulpfile.js]: Perhaps we could add a comment here.
```suggestion
// We moved the examples to the `camel-examples` repository,
// this made sure that the examples were copied to the
// correct place for the documentation, according to the
// Antora directory structure. For now the examples are
// copied over from `camel-examples` manually.
//exports.examples = examples;
```

**Comment by zregvart** [comment]
I've had a look at the examples we currently reference from the documentation. Only two pages bring in examples outside of the camel (main) repository.

The console example:

https://github.com/apache/camel/blob/e7c0a9ca73d13f9d88896eae81aafff13e49655a/docs/user-manual/modules/ROOT/pages/console-example.adoc#L16

And the thorough example:

https://github.com/apache/camel/blob/master/docs/user-manual/modules/ROOT/pages/walk-through-an-example.adoc

Other examples are taken from the core or from the components. For example the mock-component documentation includes an example from the core here:

https://github.com/apache/camel/blob/e7c0a9ca73d13f9d88896eae81aafff13e49655a/components/camel-mock/src/main/docs/mock-component.adoc#L273

Perhaps it would be better to reinstate the example files copying in `docs/gulpfile.js` and inline the examples in the above two asciidoc files.

**Comment by omarsmak** [comment]
Aha, then I shall fix this and raise another PR. Thanks @zregvart !

**Pull request merged**